libraries/abstractions/common_io/test/test_iot_tsensor.c

```diff
@@ -605,28 +638,34 @@ TEST( TEST_IOT_TSENSOR, AFQP_IotTsensorTriggerMinThreshold )
     IotTsensorHandle_t xTsensorHandle;
     int32_t lRetVal;
     int32_t setLowThreshold = -999000;
-    int32_t setHighThreshold = 999000;
 
     xTsensorHandle = iot_tsensor_open( uctestIotTsensorInstance );
     TEST_ASSERT_NOT_EQUAL( NULL, xTsensorHandle );
 
-    iot_tsensor_set_callback( xTsensorHandle, prvTsensorCallback, NULL );
-
-    lRetVal = iot_tsensor_ioctl( xTsensorHandle, eTsensorSetMinThreshold, &setHighThreshold );
-
-    if( lRetVal != IOT_TSENSOR_NOT_SUPPORTED )
+    if( TEST_PROTECT() )
     {
+        lRetVal = iot_tsensor_enable( xTsensorHandle );
         TEST_ASSERT_EQUAL( IOT_TSENSOR_SUCCESS, lRetVal );
 
-        lRetVal = xSemaphoreTake( xtestIotTsensorSemaphore, portMAX_DELAY );
-        TEST_ASSERT_EQUAL( pdTRUE, lRetVal );
-    }
+        /* Read out current temperature */
+        lRetVal = iot_tsensor_get_temp( xTsensorHandle, &setLowThreshold );
+        TEST_ASSERT_EQUAL( IOT_TSENSOR_SUCCESS, lRetVal );
 
-    lRetVal = iot_tsensor_ioctl( xTsensorHandle, eTsensorSetMinThreshold, &setLowThreshold );
+        /* Set min threshold to be 5 degrees C higher than current temperature.
+         * Expect it to fire immediately */
+        setLowThreshold += 5000;
```

Comment:
These values are in _millidegrees Celsius_, consider a naming convention so that maintainers/readers can ensure the values are used & converted properly.